test/helpers/kubectl.go

```diff
@@ -1612,6 +1612,41 @@ func (kub *Kubectl) EtcdOperatorReport(ctx context.Context, reportCmds map[strin
 	}
 }
 
+// WaitForEtcdCRDReady inspects the etcdclusters CRD to validate that the etcd
+// pods being started by the etcd-operator are ready.
+func (kub *Kubectl) WaitForEtcdCRDReady(namespace, name string, expectedCount int, timeout time.Duration) error {
+	// Test filter: https://jqplay.org/s/_rqbx7hTmI
+	jqFilter := fmt.Sprintf(`.status.members.ready | length`)
+	cmd := fmt.Sprintf("%s get etcdclusters -n %s %s -o json | jq '%s'",
+		KubectlCmd, namespace, name, jqFilter)
+	kub.logger.Infof("Querying etcdclusters on %s/%s", namespace, name)
+
+	body := func() bool {
+		res := kub.ExecShort(cmd)
```

Comment:
I think we have ExecMedium and maybe Long if you preferred longer timeouts (some comment somewhere was about timeouts). If it works, though...

Reply:
:+1: Wasn't aware of that. I think that the execution of the CRD fetch should be pretty short? This was just following the outline of a similar function elsewhere so not sure whether it's necessary to extend the timeout.